lib/wibox/layout/margin.lua

```diff
@@ -78,6 +78,14 @@ function margin:get_children()
     return {self.widget}
 end
 
+--- Replace the layout children
+-- This layout only accept one children, all others will be ignored
+-- @tparam table children A table composed of valid widgets
+function margin:set_children(children)
+    self.widget = children and children[1]
```

Comment:
This is a good point. I asked myself the same thing (and also while writing the client layout system). After thinking about it, I decided that no, it isn't worth it. My decision is debatable. `check_widget` does show up when profiling, but it isn't a bottleneck. In the end it's just that the actual implementation of `check_widget` is actually not that useful (IMO) and doesn't really enforce anything beside basic "yes, this should be a widget, at least I hope so". You can still make invalid widgets that pass the test.

I added an `is_widget` below. If this is to true and it isn't a widget, then I blame the user.